src/barretenberg_structures.rs

```diff
@@ -641,13 +648,91 @@ impl ConstraintSystem {
             buffer.extend(&constraint.to_bytes());
         }
 
+        // Serialize each Block constraint
+        let len = self.block_constraints.len() as u32;
+        buffer.extend_from_slice(&len.to_be_bytes());
+        for constraint in self.block_constraints.iter() {
+            buffer.extend(&constraint.to_bytes());
+        }
+
+        buffer
+    }
+}
+
+#[derive(Clone, Hash, Debug)]
+pub(crate) struct MemOpBarretenberg {
+    pub(crate) index: Constraint,
+    pub(crate) value: Constraint,
+    pub(crate) is_store: i8,
```

Comment:
Let's make this `is_store` the first value so it matches with the struct in barretenberg and it is clear why we are writing it first in `to_bytes`

Reply:
done